lib/ansible/modules/cloud/amazon/aws_s3.py

```diff
@@ -744,6 +760,12 @@ def main():
     if not s3_url and 'S3_URL' in os.environ:
         s3_url = os.environ['S3_URL']
 
+    if dualstack and s3_url:
+        module.fail_json(msg='dualstack only applies to AWS S3')
+
+    if dualstack and LooseVersion(botocore.__version__) < LooseVersion('1.4.45'):
```

Comment:
Instead of using LooseVersion, you can use the AnsibleAWSModule feature `module.botocore_at_least('1.4.45')` from [here](https://github.com/ansible/ansible/blob/5af1d848426d68da624c2d0cda632b8863ff34fc/lib/ansible/module_utils/aws/core.py#L218)

```
if dualstack and not module.botocore_at_least('1.4.45'):
    .....
```